tests/expressions/tests.py

```diff
@@ -534,6 +534,21 @@ def test_subquery_references_joined_table_twice(self):
         outer = Company.objects.filter(pk__in=Subquery(inner.values('pk')))
         self.assertFalse(outer.exists())
 
+    def test_subquery_filter_by_aggregate(self):
+        Number.objects.create(integer=1000, float=1.2)
+        Employee.objects.create(firstname="", lastname="", salary=1000)
```

Comment:
I should have been clearer but `firstname` and `lastname` can be omited as they'll default to `''` if missing.